tests/core_tests/chaingen.cpp

```diff
@@ -582,13 +587,15 @@ void fill_tx_sources_and_destinations(const std::vector<test_event_entry>& event
   destinations.push_back(de);
 
   tx_destination_entry de_change;
-  uint64_t cache_back = get_inputs_amount(sources) - (amount + fee);
-  if (0 < cache_back)
+  uint64_t cash_back = get_inputs_amount(sources) - (amount + fee);
+  if (0 < cash_back)
   {
-    if (!fill_tx_destination(de_change, from, cache_back))
+    if (!fill_tx_destination(de_change, from, cash_back))
       throw std::runtime_error("couldn't fill transaction cache back destination");
     destinations.push_back(de_change);
   }
+
+  if (change_amount) *change_amount = (cash_back > 0) ? cash_back : 0;
```

Comment:
boost::optional is less bug prone than null pointers @msgmaxim I actually was reminded about this when reading the rust book. I see what you mean now about compile-time reference checks, very cool

Reply:
Nice! :)